Libraries/Cordio/controller/sources/ble/init/init.c

```diff
@@ -339,3 +339,24 @@ uint32_t LlInit(LlInitRtCfg_t *pCfg)
 
   return totalMemUsed;
 }
+
+/*************************************************************************************************/
+/*!
+ *  \brief  Set LL and BB runtime configurations.
+ *
+ *  \param  pCfg    Runtime configuration.
+ *
+ *  \return Memory used.
+ */
+/*************************************************************************************************/
+uint32_t LlInitMemConfig(LlInitRtCfg_t *pCfg)
```

Comment:
I decided to name this function *MemConfig* because I see it is doing two things:
- Setting run-time configurations
- Computing the amount of heap memory used

Though, perhaps I could have simply named it `LlInitSetRtCfg()`. I think this is preferable. Let me know what you think.

Reply:
I like ```LlInitSetRtCfg()```. Take the guts of 322 to 334 and replace those lines with a call to ```LlInitSetRtCfg()```.

Reply:
### Observation

I noticed in `LlInit()` that the following lines are performed after calling both `LlInitSetBbRtCfg()` and `LlInitSetLlRtCfg()`.

```C
pCfg->pFreeMem += memUsed;
pCfg->freeMemAvail -= memUsed;
```

In the updated examples, on the other hand, these are performed only once after `LlInitSetBbRtCfg()` and omitted after `LlInitSetLlRtCfg()`. Such as, [here](https://github.com/analogdevicesinc/msdk/pull/1179/files#diff-7c5e05044eb55068c970128b55bc1b31c278e44f82c0d2325b275d4606af3d9fR127-R133).

### Conclusion

I am willing to bet that the examples needed that additional increment/decrement of `memUsed`. I will go ahead and add it to our new function `LlInitSetRtCfg()`. Is that correct?

Reply:
I don't think it's significant, but it can't hurt. I'm pretty sure that llCfg structure is only used to pass the initialization paramters. I don't think it's used once it returns from the init function.